libclc/ptx-nvidiacl/libspirv/images/image.cl

```diff
@@ -973,3 +975,1511 @@ _CLC_DEF int3 _Z22__spirv_ImageQuerySizeIDv3_i14ocl_image3d_roET_T0_(
 }
+
+// <--- BINDLESS IMAGES PROTOTYPE --->
+
+// UNSAMPLED IMAGES
+
+// Generated funcs -- READS
+// int -- int4 already defined
+int __nvvm_suld_1d_i32_clamp_s(long, int) __asm("llvm.nvvm.suld.1d.i32.clamp");
+int __nvvm_suld_2d_i32_clamp_s(long, int,
+                               int) __asm("llvm.nvvm.suld.2d.i32.clamp");
+int __nvvm_suld_3d_i32_clamp_s(long, int, int,
+                               int) __asm("llvm.nvvm.suld.3d.i32.clamp");
+int2 __nvvm_suld_1d_v2i32_clamp_s(long imageHandle, int coord) {
+  int4 ret;
+  ret = __nvvm_suld_1d_v4i32_clamp_s(imageHandle, coord);
+  int2 b;
+  b.x = ret.x;
+  b.y = ret.y;
+  return b;
+}
+int2 __nvvm_suld_2d_v2i32_clamp_s(long imageHandle, int x, int y) {
+  int4 ret;
+  ret = __nvvm_suld_2d_v4i32_clamp_s(imageHandle, x, y);
+  int2 b;
+  b.x = ret.x;
+  b.y = ret.y;
+  return b;
+}
+int2 __nvvm_suld_3d_v2i32_clamp_s(long imageHandle, int x, int y, int z) {
+  int4 ret;
+  ret = __nvvm_suld_3d_v4i32_clamp_s(imageHandle, x, y, z);
+  int2 b;
+  b.x = ret.x;
+  b.y = ret.y;
+  return b;
+}
+
+// unsigned int
+unsigned int __nvvm_suld_1d_j32_clamp_s(long imageHandle, int coord) {
+  return as_uint(__nvvm_suld_1d_i32_clamp_s(imageHandle, coord));
+}
+unsigned int __nvvm_suld_2d_j32_clamp_s(long imageHandle, int x, int y) {
+  return as_uint(__nvvm_suld_2d_i32_clamp_s(imageHandle, x, y));
+}
+unsigned int __nvvm_suld_3d_j32_clamp_s(long imageHandle, int x, int y, int z) {
+  return as_uint(__nvvm_suld_3d_i32_clamp_s(imageHandle, x, y, z));
+}
+uint2 __nvvm_suld_1d_v2j32_clamp_s(long imageHandle, int coord) {
+  return as_uint2(__nvvm_suld_1d_v2i32_clamp_s(imageHandle, coord));
+}
+uint2 __nvvm_suld_2d_v2j32_clamp_s(long imageHandle, int x, int y) {
+  return as_uint2(__nvvm_suld_2d_v2i32_clamp_s(imageHandle, x, y));
+}
+uint2 __nvvm_suld_3d_v2j32_clamp_s(long imageHandle, int x, int y, int z) {
+  return as_uint2(__nvvm_suld_3d_v2i32_clamp_s(imageHandle, x, y, z));
+}
+uint4 __nvvm_suld_1d_v4j32_clamp_s(long imageHandle, int coord) {
+  return as_uint4(__nvvm_suld_1d_v4i32_clamp_s(imageHandle, coord));
+}
+uint4 __nvvm_suld_2d_v4j32_clamp_s(long imageHandle, int x, int y) {
+  return as_uint4(__nvvm_suld_2d_v4i32_clamp_s(imageHandle, x, y));
+}
+uint4 __nvvm_suld_3d_v4j32_clamp_s(long imageHandle, int x, int y, int z) {
+  return as_uint4(__nvvm_suld_3d_v4i32_clamp_s(imageHandle, x, y, z));
+}
+
+// short -- short4 already define
+short __nvvm_suld_1d_i16_clamp_s(long,
+                                 int) __asm("llvm.nvvm.suld.1d.i16.clamp");
+short __nvvm_suld_2d_i16_clamp_s(long, int,
+                                 int) __asm("llvm.nvvm.suld.2d.i16.clamp");
+short __nvvm_suld_3d_i16_clamp_s(long, int, int,
+                                 int) __asm("llvm.nvvm.suld.3d.i16.clamp");
+short2 __nvvm_suld_1d_v2i16_clamp_s(long imageHandle, int coord) {
+  short4 ret;
+  ret = __nvvm_suld_1d_v4i16_clamp_s(imageHandle, coord);
+  short2 b;
+  b.x = ret.x;
+  b.y = ret.y;
+  return b;
```

Comment:
I agree that it could be done, however it seems to be a common pattern across the codebase ([builtins](https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/oneapi/experimental/cuda/builtins.hpp#L80)) if we introduced it in here I'd be tempted to say we should somehow generalise it and roll out everywhere?

Reply:
I think generalizing everywhere is beyond what we are trying to achieve with this PR, and as we are already pressed for time to get all our extension PRs in before feature freeze, I think we should leave this as is.